modules/images/dominant-color/class-dominant-color-image-editor-gd.php

```diff
@@ -35,12 +35,7 @@ public function get_dominant_color() {
 		$shorted_image = imagecreatetruecolor( 1, 1 );
 		imagecopyresampled( $shorted_image, $this->image, 0, 0, 0, 0, 1, 1, imagesx( $this->image ), imagesy( $this->image ) );
 
-		$hex = dechex( imagecolorat( $shorted_image, 0, 0 ) );
-
-		if ( strlen( $hex ) < 6 ) {
-			return new WP_Error( 'image_editor_dominant_color_error', __( 'Dominant color detection failed.', 'performance-lab' ) );
-		}
-		return $hex;
+		return dechex( imagecolorat( $shorted_image, 0, 0 ) );
```

Comment:
```suggestion
		$rgb = imagecolorat( $shorted_image, 0, 0 );
		$r = ($rgb >> 16) & 0xFF;
		$g = ($rgb >> 8) & 0xFF;
		$b = $rgb & 0xFF;
		return sprintf("%02x%02x%02x", $r, $g, $b);
```

Spent ages on this one. But finally got 6 digit hex values returning. Basically what this does, cover to rgb and then convert to hex from there.

Reply:
Fixed.